### PR TITLE
Add json parsing doc section for Filebeat on Kubernetes

### PR DESCRIPTION
**Pull request by ChrsMark**
Adds documentation section about json parsing for Filebeat on Kubernetes.

Closes https://github.com/elastic/beats/issues/22516

@dedemorton  do you think I should add it in the new docs guide we are preparing?

### CODE REVIEW

**Comment by Dede** [comment]
Hey @ChrsMark, i think you wanted to mention @dedemorton 😃

**Comment by ChrsMark** [comment]
> Hey @ChrsMark, i think you wanted to mention @dedemorton 😃

🤦🏼 you r right! sorry for this!

File: filebeat/docs/running-on-kubernetes.asciidoc [hunk 1]
```diff
@@ -135,3 +135,115 @@ filebeat   32        32        0         32           0           <none>
...
+                  - "/var/log/containers/*-${data.kubernetes.container.id}.log"
+          - condition:
+              contains:
+                kubernetes.container.name: "json-logging"
+            config:
+              - type: container
+                paths:
+                  - "/var/log/containers/*-${data.kubernetes.container.id}.log"
+                processors:
+                  - decode_json_fields:
```
Comment (exekias) [inline filebeat/docs/running-on-kubernetes.asciidoc:211]: Should this one still be `json.*`?

File: filebeat/docs/running-on-kubernetes.asciidoc [hunk 2]
```diff
@@ -135,3 +135,115 @@ filebeat   32        32        0         32           0           <none>
...
 Log events should start flowing to Elasticsearch. The events are annotated with
 metadata added by the <<add-kubernetes-metadata>> processor.
+
+
+[float]
+==== Parsing json logs
+
+It is common case when collecting logs from workloads running on Kubernetes that these
+applications are logging in json format. In these case, special handling can be applied so as to
+parse these json logs properly and decode them into fields. Bellow there are provided 3 different ways
```
Comment (exekias) [inline filebeat/docs/running-on-kubernetes.asciidoc:145]: I think we should be more opinionated here, adding just 2 options. One for template based autodiscover and another one for hints based autodiscover. I would go for `json.*` as it gives the user more freedom down the line (multiline/line filtering happens after JSON decoding).

**Comment by ChrsMark** [comment]
@exekias fixed, could you have another look please?

**Review by exekias — APPROVED** [review]

**Pull request merged**